quickwit/quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -478,22 +485,23 @@ impl IndexingService {
 
         let indexes_metadata_futures = new_pipeline_ids
             .iter()
-            .unique_by(|pipeline_id| pipeline_id.index_id.clone())
-            .map(|pipeline_id| self.index_metadata(ctx, &pipeline_id.index_id));
+            .unique_by(|pipeline_id| pipeline_id.index_config_id.clone())
+            .map(|pipeline_id| self.index_metadata(ctx, &pipeline_id.index_config_id.index_id));
         let indexes_metadata = try_join_all(indexes_metadata_futures).await?;
-        let indexes_metadata_by_index_id: HashMap<String, IndexMetadata> = indexes_metadata
+        let indexes_metadata_by_index_id: HashMap<IndexConfigId, IndexMetadata> = indexes_metadata
             .into_iter()
-            .map(|index_metadata| (index_metadata.index_config.index_id.clone(), index_metadata))
+            .map(|index_metadata| (index_metadata.index_config_id(), index_metadata))
             .collect();
 
         let mut failed_spawning_pipeline_ids: Vec<IndexingPipelineId> = Vec::new();
 
         // Add new pipelines.
         for new_pipeline_id in updated_pipeline_ids.difference(&running_pipeline_ids) {
```

Comment:
While you are editing this file, can you do the necessary change to reuse `new_pipeline_ids`?
The current code is weird.

```suggestion
        for new_pipeline_id in new_pipeline_ids {
```

Reply:
Let's get #3215 in and then I will see if I can make it cleaner overall.